script/lint

```diff
@@ -21,8 +21,27 @@ echo "LINT with pylint"
 echo "================"
 pylint_files=$(echo "$files" | grep -v '^tests.*')
 if [ -z "$pylint_files" ] ; then
-  echo "Only test files changed. Skipping\n"
-  exit
+  echo -e "Only test files changed. Skipping pylint\n"
+else
+  pylint $pylint_files
+fi
+echo "================"
+echo "TYPING with mypy"
```

Comment:
So is typing a part of linting? I would say no, because we also run it separately on CI.

Reply:
CC @scop

Reply:
The `script/lint` script is super convenient as a quick check before committing, the point of this PR is to make it simple to run mypy on relevant files.

Reply:
IMHO, this should be part of your IDE or any other tool of choice.

Reply:
Maybe, but isn't that true for `script/lint` already?
Anyway, maybe this is a useless PR, feel free to close it.

Reply:
I see the value of a pre-commit test, but I would rather see mypy being run as part of the actual `pre-commit` setup, not a custom script. I think it would be fast enough to be added there, but there's some work ahead with the configs before that'll work properly.

Reply:
Not that bad after all, so submitted as PR #27339

Reply:
@scop That's awesome! I'll close this PR then.